tests/schemer/sample.py

```diff
@@ -66,7 +66,8 @@ def get_website_schema(document):
     "misc":             {"type": Mixed(basestring, int)},
     "linked_id":        {"type": Mixed(int, basestring)},
     "external_code":    {"type": basestring, "nullable": False},
-    "website":          {"type": get_website_schema}
+    "website":          {"type": get_website_schema},
+    "editors":          {"type": Array(lambda document: name_schema)}
```

Comment:
Would be nice if this pointed to a separate function which returned a different type depending on some condition, to illustrate what this capability is actually good for.

Maybe have the `'editors'` list allow either a `str` for the name or a `name_schema`?